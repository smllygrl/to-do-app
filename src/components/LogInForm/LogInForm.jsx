import {
  securePassword,
  checkLength,
} from "../../helpers/password/passwordHelp";

import { validCharacters } from "../../helpers/username/usernameHelp";

const LogInForm = () => {
  // INPUT TESTING NOT SUBMIT TESTING

  const handlePasswordChange = (event) => {
    let tempPswrd = event.target.value;
    validCharacters(tempPswrd);
    checkLength(tempPswrd);
    securePassword(tempPswrd);
  };

  const handleUserNameChange = (event) => {
    let tempUserName = event.target.value;
    validCharacters(tempUserName);
  };

  return (
    <div data-testid="login-1">
      <form name="logIn">
        <label>
          Username:
          <input onChange={handleUserNameChange} />
        </label>
        <label>
          Password:
          <input
            placeholder=""
            onChange={handlePasswordChange}
            type="password"
          />
        </label>
      </form>
    </div>
  );
};

export default LogInForm;
