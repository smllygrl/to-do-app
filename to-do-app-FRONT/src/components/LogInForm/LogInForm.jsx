const LogInForm = () => {
  // This component only handles the user input and testing
  // NOT SUBMIT TESTING

  const handlePassChange = (event) => {
    let pswrd = event.target.value;
    console.log(pswrd);

    // can't contain ~ ` < >
    // length must be between 5 and 12
    if (pswrd.length >= 5) {
      // const firstFour = partial.substring(0, 4); // "pass"
      // const partialLength = partial.length;
      // const lengthOfAstericks = partialLength - 4;
      // const returnStr = firstFour + (lengthOfAstericks * "*");

      let pswrdArr = pswrd.split("");
      for (let i = 0; i < pswrd.length; i++) {
        if (i > 3) {
          pswrdArr[i] = "*";
        }
      }

      let returnStr = pswrdArr.join("");
      console.log(returnStr);
      event.innerText = returnStr;
      console.log(pswrd);
    }
    return pswrd;
  };

  return (
    <form name="logIn">
      <input
        placeholder="Username"
        // onChange={handleUserChange()}
      ></input>
      <input placeholder="Password" onChange={handlePassChange}></input>
    </form>
  );
};

export default LogInForm;
