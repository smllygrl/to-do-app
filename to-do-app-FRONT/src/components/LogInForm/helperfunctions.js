export const securePassword = (password) => {
  // password > pass****
  let passwordArr = password.split("");
  for (let i = 0; i < password.length; i++) {
    if (i > 3) {
      passwordArr[i] = "*";
    }
  }
  let returnStr = passwordArr.join("");

  return returnStr;
  // does NOT render pass**** on front end
};

export const checkLength = (password) => {
  if (password.length >= 5) {
    if (password.length < 12) {
      return true;
    }
  }
  console.log("Password must be between 5 and 12 digits");
  return false;
};

export const validCharacters = (str) => {
  // as each character comes in, we want to know it is valid

  let strArr = str.split("");

  const regEx = /([A-Za-z0-9-_]){1}/;

  for (let i = 0; i < strArr.length; i++) {
    if (strArr[i].match(regEx)) {
      continue;
    } else {
      console.log("Can only contain A-Z, a-z, 0-9, -, _");
      return false;
    }
  }
  console.log(`${str} is valid`);
  return true;
};
