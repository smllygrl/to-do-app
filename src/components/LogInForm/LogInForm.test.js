import React from "react";

import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import LogInForm from "./LogInForm";

afterEach(() => {
  cleanup();
});

test("test", () => {
  expect(true).toBe(true);
});

test("should render LogInForm component", () => {
  render(<LogInForm />);
  const loginformElem = screen.getByTestId("login-1");
  expect(loginformElem).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer.create(<LogInForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test("should accept valid password inputs", () => {
//   render(<LogInForm />);
//   const handleChange = jest.fn();
//   const passwordInput = screen.findByPlaceholderText("Password");
//   const validPassword = "Natalie09-";
//   fireEvent.change(passwordInput, validPassword);
//   expect(handleChange).toBe(true);
// });

// test("should not accept valid password inputs", () => {});

// test("should implement privacy measures on password", () => {});

// last 4 digits of password should be * when length >= 5
// can't contain ~ ` < >
// length must be between 5 and 12

// ONLY THROW REAL ERRORS ON SUBMIT
// otherwise, just display information on page
