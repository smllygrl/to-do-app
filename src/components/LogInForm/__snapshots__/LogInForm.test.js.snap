// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="login-1"
>
  <form
    name="logIn"
  >
    <input
      placeholder="Username"
    />
    <input
      onChange={[Function]}
      placeholder="Password"
    />
  </form>
</div>
`;
